src/ast/passes/codegen_llvm.cpp

```diff
@@ -927,6 +927,18 @@ void CodegenLLVM::visit(Call &call)
     if (!addr)
       throw FatalUserException("Failed to resolve kernel symbol: " + name);
     expr_ = b_.getInt64(addr);
+  } else if (call.func == "percpu_kaddr") {
+    auto name = bpftrace_.get_string_literal(call.vargs.at(0));
+    auto var = b_.CreatePointerCast(DeclareKernelVar(name), b_.GET_PTR_TY());
+    Value *percpu_ptr;
+    if (call.vargs.size() == 1) {
+      percpu_ptr = b_.CreateThisCpuPtr(var, call.loc);
+    } else {
+      auto scoped_del = accept(call.vargs.at(1));
+      Value *cpu = expr_;
+      percpu_ptr = b_.CreatePerCpuPtr(var, cpu, call.loc);
+    }
+    expr_ = b_.CreatePtrToInt(percpu_ptr, b_.getInt64Ty());
```

Comment:
Is pointer to u64 right here? semantic analyser seems to mark the return type as u64

Reply:
It creates a cast from a ptr to an integer, not to a "pointer to integer".